pkg/auth/auth_token_handler.go

```diff
@@ -141,13 +145,74 @@ func (a *AuthTokenHandlerImpl) refreshCCloudSSOToken(client *ccloudv1.Client, re
 	return res.GetToken(), refreshToken, err
 }
 
-func (a *AuthTokenHandlerImpl) GetConfluentToken(mdsClient *mdsv1.APIClient, credentials *Credentials) (string, error) {
+func (a *AuthTokenHandlerImpl) GetConfluentToken(mdsClient *mdsv1.APIClient, credentials *Credentials, noBrowser bool) (string, string, error) {
 	ctx := utils.GetContext()
-	basicContext := context.WithValue(ctx, mdsv1.ContextBasicAuth, mdsv1.BasicAuth{UserName: credentials.Username, Password: credentials.Password})
-	resp, _, err := mdsClient.TokensAndAuthenticationApi.GetToken(basicContext)
+	if credentials.IsSSO {
+		if credentials.AuthRefreshToken != "" {
+			if authToken, err := refreshConfluentToken(mdsClient, credentials); err == nil {
+				return authToken, "", nil
+			}
+		}
+
+		resp, _, err := mdsClient.SSODeviceAuthorizationApi.Security10OidcDeviceAuthenticatePost(context.Background())
+		if err != nil {
+			return "", "", err
+		}
+
+		if noBrowser {
+			output.Println(false, "Navigate to the following link in your browser to authenticate:")
+			output.Println(false, resp.VerificationUri)
+		} else {
+			if err := browser.OpenURL(resp.VerificationUri); err != nil {
+				return "", "", fmt.Errorf("unable to open web browser for SSO authentication: %w", err)
+			}
+		}
+
+		checkDeviceAuthRequest := mdsv1.CheckDeviceAuthRequest{
+			UserCode: resp.UserCode,
+			Key:      resp.Key,
+		}
+
+		var authToken, refreshToken string
+		err = retry.Retry(time.Duration(resp.Interval)*time.Second, time.Duration(resp.ExpiresIn)*time.Second, func() error {
+			checkAuthResp, _, err := mdsClient.SSODeviceAuthorizationApi.CheckDeviceAuth(context.Background(), checkDeviceAuthRequest)
+			if err != nil {
+				return fmt.Errorf("%s: %w", checkAuthResp.Error, err)
+			}
+
+			if !checkAuthResp.Complete {
+				return fmt.Errorf("%s: %s", checkAuthResp.Status, checkAuthResp.Description)
+			}
+
+			authToken = checkAuthResp.AuthToken
+			refreshToken = checkAuthResp.RefreshToken
+			return nil
+		})
+		if err != nil {
+			return "", "", err
+		}
+
+		return authToken, refreshToken, nil
+	} else {
+		basicContext := context.WithValue(ctx, mdsv1.ContextBasicAuth, mdsv1.BasicAuth{UserName: credentials.Username, Password: credentials.Password})
```

Comment:
Let's construct `basicAuth` outside of this function call. This line is a bit too long